cpp/CMakeLists.txt

```diff
@@ -537,7 +553,9 @@ target_compile_options(
                "$<$<COMPILE_LANGUAGE:CUDA>:${CUVS_CUDA_FLAGS}>"
 )
 # ensure CUDA symbols aren't relocated to the middle of the debug build binaries
-target_link_options(cuvs PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld")
+#
+# TODO(achirkin): disabled during experiments with CUDA_SEPARABLE_COMPILATION (otherwise did't link)
+# target_link_options(cuvs PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld")
```

Comment:
Uncommenting this back leads to an error duing linking phase:
```
nvcc -Xcompiler=cpp/build/fatbin.ld -Xcompiler=-fPIC -shared -dlink <bunch_of.o> -o CMakeFiles/cuvs.dir/cmake_device_link.o <a_few_of.a> -lpthread -ldl -lcudadevrt -lcudart
x86_64-conda-linux-gnu-c++: fatal error: cannot specify '-o' with '-c', '-S' or '-E' with multiple files
compilation terminated.
```
@robertmaynard would you have a moment to help with this? The main change I introduced here was to factor out a part of the library into a static component with `CUDA_SEPARABLE_COMPILATION` flag enabled.

Reply:
Try wrapping it in a `$<HOST_LINK: <link flag> >` generator expression

Reply:
Seems to compile, thanks!